src/FWlib/peripheral_gpio.c

```diff
@@ -325,4 +325,10 @@ void GIO_ClearBits(const uint64_t index_mask)
     APB_GPIO1->DoutClear |= index_mask >> 21;
 }
 
+void GIO_ToggleBits(const uint64_t index_mask)
+{
+    APB_GPIO0->DataOut = APB_GPIO0->DataOut ^ index_mask;
+    APB_GPIO1->DataOut = APB_GPIO1->DataOut ^ index_mask >> 21;
```

Comment:
Operator precedence is hard to remember. Suggestion: `APB_GPIO1->DataOut ^ (index_mask >> 21)`